swap/src/bitcoin/wallet.rs

```diff
@@ -353,55 +350,69 @@ fn estimate_fee(
     transfer_amount: Amount,
     fee_rate: FeeRate,
     min_relay_fee: Amount,
-) -> Amount {
-    // Doing some heavy math here :)
-    // `usize` is 32 or 64 bits wide, but `f32`'s mantissa is only 23 bits wide.
-    // This is fine because such a big transaction cannot exist and there are also
-    // no negative fees.
-    #[allow(
-        clippy::cast_precision_loss,
-        clippy::cast_possible_truncation,
-        clippy::cast_sign_loss
-    )]
-    let sats_per_vbyte = ((weight as f32) / 4.0 * fee_rate.as_sat_vb()) as u64;
+) -> Result<Amount> {
+    if transfer_amount.as_sat() <= 546 {
+        bail!("Amounts needs to be greater than Bitcoin dust amount.")
+    }
+    if fee_rate.as_sat_vb() <= 0.0 {
+        bail!("Fee rate needs to be > 0")
+    }
+    if fee_rate.as_sat_vb() >= 100_000_000.0 || min_relay_fee.as_sat() > 100_000_000 {
+        bail!("A fee_rate or min_relay_fee of > 1BTC does not make sense")
+    }
```

Comment:
Did we just not have these checks before or am I missing something?

Reply:
You are not missing anything and there were no checks in place :) I added these when I added proptest-tests as I realized really high values simply do not make sense.